### PR TITLE
add google sheets connector template

### PR DESCRIPTION
**Pull request by mrebrii**
## Description

Add new template for Google Sheets connector

https://github.com/camunda/connectors-bundle/assets/131355515/264510f8-68ce-4c98-80d0-8cb12e35cd3d

## Related issues

#555 

closes #

### CODE REVIEW

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (igpetrov) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:3]: Let's give it a proper name

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (igpetrov) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:55]:
```suggestion
          "name": "Bearer token",
```

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (igpetrov) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:59]:
```suggestion
          "name": "Refresh token",
```

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (igpetrov) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:70]:
```suggestion
      "label": "Bearer token",
```

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (igpetrov) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:87]:
```suggestion
      "label": "Client ID",
```

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (igpetrov) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:88]:
```suggestion
      "description": "Enter Google API Client ID",
```

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (igpetrov) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:105]:
```suggestion
      "description": "Enter Google API client secret",
```

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (igpetrov) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:121]:
```suggestion
      "label": "Refresh token",
```

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (igpetrov) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:122]:
```suggestion
      "description": "Enter Google API refresh token that is authorized to interact with desired Drive",
```

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (igpetrov) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:250]:
```suggestion
      "label": "Worksheet ID",
```

Comment (igpetrov) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:250]:
```suggestion
      "label": "Worksheet ID",
```

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (igpetrov) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:251]:
```suggestion
      "description": "Enter the ID of the worksheet",
```

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (igpetrov) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:274]:
```suggestion
      "label": "Worksheet ID",
```

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (igpetrov) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:275]:
```suggestion
      "description": "Enter the ID of the worksheet",
```

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (igpetrov) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:551]:
```suggestion
      "label": "Result variable",
```

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (igpetrov) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:561]:
```suggestion
      "label": "Result expression",
```

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (igpetrov) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:572]:
```suggestion
      "label": "Error expression",
```

**Review by igpetrov — CHANGES_REQUESTED** [review]
LGTM, several changes proposed to normalize look&feel

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:55]: It's the same as in Google drive connector. I think the name should be consistent with it . So, we should define the right name and change it in Drive connector too if it's necessary

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:59]: the same topic as in the previous conversation

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:70]: the same discussion is needed

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:87]: the same one

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:88]: the same one

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:104]: the same one

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:105]: the same one

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:121]: the same one

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:122]: the same one

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:3]: fixed

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:250]: fixed

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:551]: part of the discussion too

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:561]: the same one

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:572]: the same one

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (igpetrov) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:55]: @mrebrii 

Let's make it as suggested and change Google Docs change as a follow up task. Do you mind creating one?

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (bastiankoerber) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:492]: can you add here an example? This makes it easier for the enduser

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (bastiankoerber) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:319]: what happens if i leave it blank? Will we add the values to the first sheet?

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (bastiankoerber) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:274]: replace with Worksheet ID

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (bastiankoerber) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:275]: replade id with ID

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (bastiankoerber) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:444]: can you add here an example?

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (bastiankoerber) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:468]: can you add here an example

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (bastiankoerber) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:457]: why do we allow secrets here?

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (bastiankoerber) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:481]: why do we allow secrets here?

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (bastiankoerber) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:512]: can you add here an example, that the user don't need to lookup the format

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (bastiankoerber) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:420]: can you add example here

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (bastiankoerber) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:372]: can you add an example here

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (bastiankoerber) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:188]: i would expect, that i can define the worksheetname here

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (bastiankoerber) [inline connectors/google-sheets/element-templates/google-sheets-connector.json:144]: please use the title case for all properties:
Instead of "Create spreadsheet" use "Create Spreadsheet"

**Review by bastiankoerber — CHANGES_REQUESTED** [review]
hi @mrebrii ,
some minor changes.

Please check all properties names, that they are "Title Case".
We should be here consistent with all other OOTB Connectors :-)

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:55]: Yes, sure

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:492]: Honestly, I've thought to do it while implementing this operation, as I will able to see how it looks better from user side

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:319]: Yes, You are right. I'm going to add more info in the documentation and than add the link

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (bastiankoerber) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:468]: a short example is always helpfull and a faster solution for our enduser. A good example is 
image

here we are saying, what we are expecting and referencing the documentation for follow up questions

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (bastiankoerber) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:444]: a short example is always helpfull and a faster solution for our enduser. A good example is
image

here we are saying, what we are expecting and referencing the documentation for follow up questions

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (bastiankoerber) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:492]: fine for me, just close this comment as soon as you have a solution :-)

Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:492]: For now, I close it, but I'll keep in mind this suggestion

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:420]: As we've decided It will be done with adding the documentation

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:444]: Add link to the documentation where it'll be described

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:468]: Add link to the documentation where it'll be described
image

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:512]: image

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:420]: image
Added link

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:372]: image

File: connectors/google-sheets/element-templates/google-sheets-connector.json
Comment (mrebrii) [reply connectors/google-sheets/element-templates/google-sheets-connector.json:457]: Remove it

**Comment by mrebrii** [comment]
@igpetrov @bastiankoerber here is the video with fixed connector template considered all provided changes

https://github.com/camunda/connectors-bundle/assets/131355515/dae4482a-d2d7-4abb-9cf5-ea71b09a2968

**Comment by mrebrii** [comment]
@bastiankoerber here is the video of connector template with parent folder property when operation "create spreadsheet"
and also with removed "create column headers" operation

https://github.com/camunda/connectors-bundle/assets/131355515/50200301-dab9-45e4-b3f1-30aa4829326b

**Review by igpetrov — APPROVED** [review]
From my side, looks good. Please wait for approval from @bastiankoerber before merge

**Review by bastiankoerber — APPROVED** [review]

**Pull request merged**
